[FIX] [16.0] purchase_allowed_product: Improve UX

Shide opened the pull request:
- Improved UX: Move field to the partner group of the form. Lines (purchase and move) has more space to see more content
- Make use_only_supplied_product field computed, stored and not readonly
- Propagation of use_only_supplied_product field to invoices
- `use_only_supplied_product` field always visible on Purchase Orders (check comments)

https://www.loom.com/share/6987c27348134673a7cae67c8ca43920?sid=7d980b6f-0917-4f7e-973c-1dac484d423e

MT-4264 @moduon @rafaelbn @fcvalgar @Gelojr plz review :)

Shide committed:
[FIX] purchase_allowed_product: Improve UX

- Make use_only_supplied_product field computed, stored and not readonly
- Propagation of use_only_supplied_product field to invoices

fcvalgar reviewed (approved):
Functional review,

LGTM thank you man @Shide 

![image](https://github.com/OCA/purchase-workflow/assets/17847432/18e4e262-7441-48c1-beba-4815ccb6f6f8)


![image](https://github.com/OCA/purchase-workflow/assets/17847432/06155795-c763-4680-b2c4-290aa2203ecc)

Gelojr reviewed (approved):
LGTM https://www.loom.com/share/9b2d2feaca90429b89474dda395e68db?sid=a1b1e212-644e-4835-acae-321bb0ed182f
As a suggestion, when entering to validate delivery notes and returning through the breadcrumb trail, the button disappears. ¿Is OK? https://www.loom.com/share/8fff4032c3d2452388760430a6de753a?sid=2edce4c7-d913-4e0a-8b75-2eb56ae67ba5

yostashiro commented:
> As a suggestion, when entering to validate delivery notes and returning through the breadcrumb trail, the button disappears. ¿Is OK? https://www.loom.com/share/8fff4032c3d2452388760430a6de753a?sid=2edce4c7-d913-4e0a-8b75-2eb56ae67ba5

I guess it's not within the scope of this PR, but indeed, it'd make sense to make the field visible for confirmed orders as well since you can still add products to those unless configured otherwise.

Shide committed:
[FIX] purchase_allowed_product: Check always visible on Purchase Orders

You always can add products to purchase order, so makes sense to be able to see the check/switch always

Shide commented:
@Gelojr @yostashiro makes totally sense. Done in a separate commit.

yostashiro reviewed (approved):
LGTM. 👍

yajo reviewed (approved):
/ocabot merge minor
